Tests/VisualRecognitionV3Tests/VisualRecognitionTests.swift

```diff
@@ -1130,4 +1236,253 @@ class VisualRecognitionTests: XCTestCase {
+                XCTAssertEqual(collection.collectionID, self.collectionID)
+                XCTAssertEqual(collection.name, self.collectionName)
+                XCTAssertNotNil(collection.images)
+                XCTAssertEqual(collection.status, "available")
+                XCTAssertEqual(collection.capacity, "1000000")
+            }
+            expectation.fulfill()
+        }
+        waitForExpectations()
+    }
+    
+    /** Retrieve test collection details. */
+    func testRetrieveCollectionDetails() {
+        let description = "Retrieve test collection."
+        let expectation = self.expectation(description: description)
+
+        visualRecognition.retrieveCollectionDetails(
+            collectionID: collectionID!,
+            failure: failWithError) { collection in
+                
+                XCTAssertEqual(collection.collectionID, self.collectionID)
+                XCTAssertEqual(collection.name, self.collectionName)
+                XCTAssertNotNil(collection.images)
+                XCTAssertEqual(collection.status, "available")
+                XCTAssertEqual(collection.capacity, "1000000")
+                
+                expectation.fulfill()
+        }
+        waitForExpectations()
+    }
+
+    /** Add images to collection. */
+    func testAddDeleteImageToCollection() {
+        let description = "Add image to test collection."
+        let expectation = self.expectation(description: description)
+
+        var imageID: String?
+        let imageFile = "obama.jpg"
+        
+        visualRecognition.addImageToCollection(
+            collectionID: collectionID!,
+            imageFile: obama,
+            failure: failWithError) { collectionImages in
+                
+                XCTAssertEqual(1, collectionImages.imagesProcessed)
+                for image in collectionImages.collectionImages {
+                    if image.imageFile == imageFile {
+                        imageID = image.imageID
+                        expectation.fulfill()
+                        return
+                    }
+                }
+                XCTFail("Image was not successfully added to the collection.")
+        }
+        waitForExpectations()
+        
+        let description2 = "Check that our image can be retrieved."
+        let expectation2 = self.expectation(description: description2)
+        
+        guard let imageIDToDelete = imageID else {
+            XCTFail("Image failed to be added to collection.")
+            return
+        }
+        visualRecognition.listImageDetails(
+            inCollection: collectionID!,
+            imageID: imageIDToDelete,
+            failure: failWithError) { collectionImage in
+                if collectionImage.imageFile == imageFile {
+                    expectation2.fulfill()
+                    return
+                }
+                XCTFail("The added image could not be retreived from the collection.")
+        }
+        waitForExpectations()
```

Comment:
Maybe we can take out this call to `listImageDetails`. We shouldn't need to manually confirm that the `addImageToCollection` call is successful, because our failure callback should have been called if it had failed.